### PR TITLE
allow dependency up downgrade for ocp peer dependencies

### PR DESCRIPTION
**Pull request by kesselb**
Example with patch: https://github.com/nextcloud/mail/actions/runs/5083890915/jobs/9135458154?pr=8496

image

Example without patch: https://github.com/nextcloud/social/actions/runs/5082148997/jobs/9131503707

image

### CODE REVIEW

File: workflow-templates/psalm-matrix.yml [hunk 1]
```diff
@@ -24,7 +24,7 @@ jobs:
       # do not stop on another job's failure
       fail-fast: false
       matrix:
-        ocp-version: [ 'dev-master', 'dev-stable26', 'dev-stable25', 'dev-stable24' ]
+        ocp-version: [ 'dev-master', 'dev-stable27', 'dev-stable26', 'dev-stable25', 'dev-stable24' ]
```
Comment (nickvergessen) [inline workflow-templates/psalm-matrix.yml:27]: 24 is eol, so could remove while on it

Comment (kesselb) [reply workflow-templates/psalm-matrix.yml]: sure!

Comment (kesselb) [reply workflow-templates/psalm-matrix.yml]: dropped the commit for 27/24 ;)

**Review by skjnldsv — APPROVED** [review]

File: workflow-templates/psalm-matrix.yml [hunk 2]
```diff
@@ -44,7 +44,7 @@ jobs:
         run: composer i
 
       - name: Install dependencies
-        run: composer require --dev nextcloud/ocp:${{ matrix.ocp-version }} --ignore-platform-reqs
+        run: composer require --dev nextcloud/ocp:${{ matrix.ocp-version }} --ignore-platform-reqs --with-dependencies
```
Comment (nickvergessen) [inline workflow-templates/psalm-matrix.yml:47]: For me this still fails with:
> Use the option --with-all-dependencies (-W) to allow upgrades, downgrades and removals for packages currently locked to specific versions.

Comment (nickvergessen) [reply workflow-templates/psalm-matrix.yml:47]: https://github.com/nextcloud/announcementcenter/actions/runs/5119157822/jobs/9204060224?pr=665

Comment (nickvergessen) [reply workflow-templates/psalm-matrix.yml:47]: Okay, also doesn't seem to work

**Pull request merged**